### PR TITLE
Add curvilinear grid support to to_cdms2 and fix mask bug

### PR DESCRIPTION
**Pull request by stefraynaud**
- [x] Tests added (for all bug fixes or enhancements)
 - [x] Tests passed (for all non-documentation changes)
 - [x] Fully documented, including ``whats-new.rst`` for all changes and api.rst for new API (remove if this change should not be visible to users, e.g., if it is an internal clean-up, or if this is part of a larger project that will be documented later)

### CODE REVIEW

File: xarray/tests/test_dataarray.py [hunk 1]
```diff
@@ -2940,6 +2942,39 @@ def test_to_and_from_cdms2(self):
         roundtripped = DataArray.from_cdms2(actual)
         assert_identical(original, roundtripped)
```
Comment (shoyer) [inline xarray/tests/test_dataarray.py]: Could you add separate test methods for this? It makes things a little easier to debug when things go wrong :).

File: xarray/convert.py [hunk 2]
```diff
@@ -38,15 +39,28 @@ def from_cdms2(variable):
     """
     values = np.asarray(variable)
     name = variable.id
-    coords = [(v.id, np.asarray(v),
-               _filter_attrs(v.attributes, cdms2_ignored_attrs))
-              for v in variable.getAxisList()]
+    dims = variable.getAxisIds()
+    coords = {}
```
Comment (shoyer) [inline xarray/convert.py]: Can you switch this to `OrderedDict` instead? We usually try to use those to keep the order of coordinates more predictable (though it's increasingly less relevant with ordered dicts by default on Python 3).

**Comment by stefraynaud** [comment]
Is it up to me to resolve the whats-new.rst conflict?

**Comment by fmaussion** [comment]
> Is it up to me to resolve the whats-new.rst conflict?

I just took care of it: you'll have to pull from origin if you want to add further changes though

**Comment by stefraynaud** [comment]
Thanks @fmaussion

**Comment by shoyer** [comment]
thanks @stefraynaud !

**Pull request merged**